rewriter/Minitest.cc

```diff
@@ -112,18 +112,51 @@ unique_ptr<ast::Expression> addSigVoid(unique_ptr<ast::Expression> expr) {
 }
 } // namespace
 
-unique_ptr<ast::Expression> recurse(core::MutableContext ctx, unique_ptr<ast::Expression> body);
+unique_ptr<ast::Expression> recurse(core::MutableContext ctx, unique_ptr<ast::Expression> body,
+                                    optional<ast::Expression *> context);
 
-unique_ptr<ast::Expression> prepareBody(core::MutableContext ctx, unique_ptr<ast::Expression> body) {
-    body = recurse(ctx, std::move(body));
+unique_ptr<ast::Expression> prepareBody(core::MutableContext ctx, unique_ptr<ast::Expression> body,
+                                        optional<ast::Expression *> context) {
+    body = recurse(ctx, std::move(body), context);
 
     auto bodySeq = ast::cast_tree<ast::InsSeq>(body.get());
     if (bodySeq) {
         for (auto &exp : bodySeq->stats) {
-            exp = recurse(ctx, std::move(exp));
+            exp = recurse(ctx, std::move(exp), context);
         }
 
-        bodySeq->expr = recurse(ctx, std::move(bodySeq->expr));
+        bodySeq->expr = recurse(ctx, std::move(bodySeq->expr), context);
+    }
+    return body;
+}
+
+void checkItBlock(core::MutableContext ctx, ast::Expression *exp) {
+    auto send = ast::cast_tree<ast::Send>(exp);
+    if (send) {
+        if (send->fun == core::Names::it() && send->args.size() == 1 && send->block != nullptr) {
+            return;
+        }
+    }
+
+    if (auto e = ctx.state.beginError(exp->loc, core::errors::Rewriter::NonItInTestEach)) {
+        e.setHeader("Only `{}` blocks are allowed inside `{}`", "it", "test_each");
+    }
+}
+
+unique_ptr<ast::Expression> prepareTestEachBody(core::MutableContext ctx, unique_ptr<ast::Expression> body,
+                                                optional<ast::Expression *> context) {
+    auto bodySeq = ast::cast_tree<ast::InsSeq>(body.get());
+    if (bodySeq) {
```

Comment:
Non-blocking preference:

1. Either put the assignment and `if` on one line (`if (auto bodySeq ...)`), or
2. Compare against `!= nullptr` in the condition.